crates/core/tedge_api/src/entity_store.rs

```diff
@@ -184,14 +179,63 @@ impl EntityStore {
             twin_data: Map::new(),
         };
 
-        Some(EntityStore {
+        let message_log = MqttMessageLog::new(log_path)?;
+
+        let mut entity_store = EntityStore {
+            mqtt_schema,
             main_device: main_device.topic_id.clone(),
             entities: HashMap::from([(main_device.topic_id.clone(), metadata)]),
             entity_id_index: HashMap::from([(entity_id, main_device.topic_id)]),
             external_id_mapper: Box::new(external_id_mapper_fn),
             external_id_validator_fn: Box::new(external_id_validator_fn),
             default_service_type,
-        })
+            message_log,
+        };
+
+        entity_store.load_from_message_log()?;
+
+        Ok(entity_store)
+    }
+
+    pub fn load_from_message_log(&mut self) -> Result<(), InitError> {
```

Comment:
Have you consider to persist not a log of registration messages but the entity store itself? Or more precisely its main component aka `entities: HashMap<EntityTopicId, EntityMetadata>`. I think this would be simpler.

Reply:
My initial plan was to persist the `EntityMetadata` struct itself. But then switched to raw `MqttMessage`s to cover other metadata messages as well, like `CommandMetadata` message which is not part of the entity store at the moment. But then I thought it'd be even better if we decide to persist the telemetry messages (the early message cache) as well in the near future.

Reply:
As discussed and agreed offline, we're sticking with the persistence of raw MqttMessages for now. But, this persistence impl won't be included in the current PR anymore. It will be added in a follow-up PR.